packages/@react-aria/grid/src/useGridCell.ts

```diff
@@ -205,18 +249,101 @@ export function useGridCell<T, C extends GridCollection<T>>(props: GridCellProps
         break;
       }
       case 'ArrowUp':
-      case 'ArrowDown':
+      case 'ArrowDown': {
+        if (keyboardNavigationBehavior === 'tab') {
+          break;
+        };
+
         // Prevent this event from reaching cell children, e.g. menu buttons. We want arrow keys to navigate
         // to the cell above/below instead. We need to re-dispatch the event from a higher parent so it still
         // bubbles and gets handled by useSelectableCollection.
-        if (!e.altKey && ref.current.contains(e.target as Element)) {
+        if (keyboardNavigationBehavior === 'arrow' && !e.altKey && ref.current.contains(e.target as Element)) {
           e.stopPropagation();
           e.preventDefault();
-          ref.current.parentElement.dispatchEvent(
-            new KeyboardEvent(e.nativeEvent.type, e.nativeEvent)
-          );
+
+          if (isEditing) {
+            ref.current.dispatchEvent(
+              new KeyboardEvent(e.nativeEvent.type, {...e.nativeEvent, key: e.key === 'ArrowUp' ? 'ArrowLeft' : 'ArrowRight'})
+            );
+          } else {
+            ref.current.parentElement.dispatchEvent(
+              new KeyboardEvent(e.nativeEvent.type, e.nativeEvent)
+            );
+          }
+        }
+        break;
+      }
+      case 'F2':
+      case 'Enter': {
+        if (isEditable) {
+          let focusable = walker.nextNode() as FocusableElement;
+
+          if (focusable && document.activeElement === ref.current) {
+            state.setEditingKey(node.key);
```

Comment:
Do we want to announce entering and exiting?

Reply:
gotcha, we can certainly play around with the announcements there then. Hopefully there is something simple we can do since in my experience using the LiveAnnouncer we have is a bit tricky due to the possibility of interrupting native announcements that the screen reader may be trying to issue/the inconsistent behavior of screenreaders across browser/screenreader combinations

Reply:
I finished laying the groundwork for this in `useGridEditAnnouncement()` until we have a proper idea of what and when to announce 👍 Would really appreciate guidance here, especially in terms of localization!